src/hyperactive/tests/test_all_objects.py

```diff
@@ -348,3 +348,109 @@ def test_gfo_integration(self, object_instance):
         assert isinstance(best_params, dict), "Best parameters should be a dictionary"
         assert "C" in best_params, "Best parameters should contain 'C'"
         assert "gamma" in best_params, "Best parameters should contain 'gamma'"
+
+    def test_selection_direction_backend(self, object_instance):
+        """Backends return argmax over standardized scores on controlled setup.
+
+        This verifies the maximization direction using a tiny, deterministic
+        experiment and a deliberately poor warm start. It is scoped per-backend
+        to avoid brittle stochastic behavior.
+        """
+        # Import backend bases to check optimizer type
+        from hyperactive.opt._adapters._gfo import _BaseGFOadapter
+        from hyperactive.opt._adapters._base_optuna_adapter import _BaseOptunaAdapter
+        from hyperactive.opt.gridsearch._sk import GridSearchSk
+        from hyperactive.opt.random_search import RandomSearchSk
+
+        # set up a simple deterministic experiment with clear best vs worst
+        from hyperactive.experiment.bench import Ackley
+
+        exp = Ackley(d=2)
+        # ackley is lower-better on evaluate; score flips sign to higher-better
+        poor = {"x0": 4.0, "x1": 4.0}
+        good = {"x0": 0.0, "x1": 0.0}
+
+        # Helper: assert that returned params equal the known good point
+        def _assert_good(best_params):
+            assert isinstance(best_params, dict)
+            assert best_params == good, (
+                f"Optimizer should select argmax of standardized score. "
+                f"Expected {good}, got {best_params}."
+            )
+
```

Comment:
I think, you can avoid lots of repetition by using `set_params`, i.e., `inst = object_instance.clone().set_params({"experiment": exp, "param_space": see_below})`.

Besides this, can we avoid hard-coding a lot of these parameters per estimator? This is not too extensible. It is fine if we use it primarily for checking the sign, but I wonder whether we can avoid all the hard coding.